### ISSUE TITLE
Segmentation fault when running gen_nn_ops.fractional_avg_pool

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59163]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
segfault happens with negative list elements.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_nn_ops
try:
  arg_0_tensor = tf.random.uniform([5, 20, 30, 3], dtype=tf.float64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_0 = 2
  arg_1_1 = -5.267949192431123
  arg_1_2 = -52.58578643762691
  arg_1_3 = 1
  arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
  arg_2 = True
  arg_3 = True
  deterministic = True
  seed = 87654321
  seed2 = 341261001
  out = gen_nn_ops.fractional_avg_pool(arg_0,arg_1,arg_2,arg_3,deterministic=deterministic,seed=seed,seed2=seed2,)
except Exception as e:
  print("Error:"+str(e))
```

```
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_nn_ops
try:
  arg_0_tensor = tf.random.uniform([1, 10, 10, 1], dtype=tf.float64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_0 = True
  arg_1_1 = -0.35668935305391647
  arg_1_2 = -0.7209753581353426
  arg_1_3 = -87
  arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
  arg_2 = True
  arg_3 = True
  deterministic = True
  seed = 87654321
  seed2 = 341261001
  out = gen_nn_ops.fractional_avg_pool(arg_0,arg_1,arg_2,arg_3,deterministic=deterministic,seed=seed,seed2=seed2,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-07 13:44:10.489552: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.493914: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.494017: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.494307: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-07 13:44:10.494924: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.495025: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.495113: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.840688: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.840834: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.840928: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-07 13:44:10.841010: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 4263 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1660 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} Fractional average pooling is not yet supported on the batch nor channel dimension. [Op:FractionalAvgPool]
Error:{{function_node __wrapped__FractionalAvgPool_device_/job:localhost/replica:0/task:0/device:CPU:0}} Both seed and seed2 should be 0 if deterministic is false. [Op:FractionalAvgPool]
Error:Expected bool for argument 'pseudo_random' not -69.0.
Error:Value for attr 'T' of uint32 is not in the list of allowed values: float, double, int32, int64
	; NodeDef: {{node FractionalAvgPool}}; Op<name=FractionalAvgPool; signature=value:T -> output:T, row_pooling_sequence:int64, col_pooling_sequence:int64; attr=pooling_ratio:list(float),min=4; attr=pseudo_random:bool,default=false; attr=overlapping:bool,default=false; attr=deterministic:bool,default=false; attr=seed:int,default=0; attr=seed2:int,default=0; attr=T:type,allowed=[DT_FLOAT, DT_DOUBLE, DT_INT32, DT_INT64]> [Op:FractionalAvgPool]
Segmentation fault

```
```
</details>

**Comment by tilakrayal** [comment]
@nimashiri,
I tried to execute the mentioned code and the issue was fixed in the latest stable [v2.11](https://colab.research.google.com/gist/tilakrayal/f89a83a8a60c96ae738915eb9035962b/untitled847.ipynb) and [nightly](https://colab.research.google.com/gist/tilakrayal/b4bdf4234828555f3e472233043a0ece/untitled847.ipynb). Kindly find the gist of it here.

**Comment by tilakrayal** [comment]
@nimashiri,
As it was the bug on tensorflow v2.10, it has been resolved on the latest stable tensorflow v2.11 where we can observe that Segfault error has not been raised. Thank you!

**Comment by nimashiri** [comment]
Also this one:

```
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_nn_ops
try:
  arg_0_tensor = tf.random.uniform([5, 20, 30, 3], dtype=tf.float64)
  arg_0 = tf.identity(arg_0_tensor)
  arg_1_0 = 1
  arg_1_1 = -15.267949192431123
  arg_1_2 = 0.0
  arg_1_3 = -1
  arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
  arg_2 = True
  arg_3 = False
  deterministic = True
  seed = 87654319
  seed2 = 341261001
  out = gen_nn_ops.fractional_avg_pool(arg_0,arg_1,arg_2,arg_3,deterministic=deterministic,seed=seed,seed2=seed2,)
except Exception as e:
  print("Error:"+str(e))
```

**Comment by tilakrayal** [comment]
> Also this one:
> 
> ```
> import tensorflow as tf
> import os
> import numpy as np
> from tensorflow.python.ops import gen_nn_ops
> try:
>   arg_0_tensor = tf.random.uniform([5, 20, 30, 3], dtype=tf.float64)
>   arg_0 = tf.identity(arg_0_tensor)
>   arg_1_0 = 1
>   arg_1_1 = -15.267949192431123
>   arg_1_2 = 0.0
>   arg_1_3 = -1
>   arg_1 = [arg_1_0,arg_1_1,arg_1_2,arg_1_3,]
>   arg_2 = True
>   arg_3 = False
>   deterministic = True
>   seed = 87654319
>   seed2 = 341261001
>   out = gen_nn_ops.fractional_avg_pool(arg_0,arg_1,arg_2,arg_3,deterministic=deterministic,seed=seed,seed2=seed2,)
> except Exception as e:
>   print("Error:"+str(e))
> ```

This issue was already tracking with #59198. And the initial issue for Segmentation fault was not able to reproduce in tf v2.11 as mentioned in the above comment. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/8d201599011ad486bc16967294da4ccb/untitled847.ipynb) where we can observe that Segfault error has not been raised and executed without any error and let us know if it serves the problem. Thank you!

**Comment by mihaimaruseac** [comment]
Looks like an issue on cleanup side.

@learning-to-play

**Comment by bharatjetti** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The code snippet provided has returned the error which is catch through the exception, and doesn't crash with latest TensorFlow version. Please find the gist [here](https://colab.sandbox.google.com/gist/bharatjetti/b97f12a2dfefef5f80b06196e8daadbe/untitled847.ipynb).

The TensorFlow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
